### PR TITLE
Added support for transitions, internationalization, dynamic updating…

### PR DESCRIPTION
**Pull request by tariq-9484**
… of purchase order total price

### CODE REVIEW

File: app/controllers/medicines_controller.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
+class MedicinesController < ApplicationController
+
+	include ActionController::MimeResponds
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb]: Fix indentation throughout your files, it should be 2 spaces, you can set it to do so by default on sublime.

File: app/controllers/medicines_controller.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          format.js{ render 'purchase_order/search' }
+      	  end
+		end
+	end
+
+	def create
+
+	  @medicine = Medicine.new(medicine_params)
+	  #todo: fix this when PR is merged
+	  @medicine.hospital=Hospital.first
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb]: This can be fixed now

File: app/controllers/medicines_controller.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	      redirect_to @medicine
+	    else   
+	      flash[:error] = t('medicine.update.failure')     
+	      render :edit   
+	    end   
+	end
+
+	def destroy
+	  @medicine = Medicine.find_by(sequence_num: params[:id]) 
+	    if @medicine.delete
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb]:
```suggestion
	    if @medicine.destroy
```

File: app/controllers/medicines_controller.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	end
+
+	def destroy
+	  @medicine = Medicine.find_by(sequence_num: params[:id]) 
+	    if @medicine.delete   
+	      flash[:notice] = t('medicine.delete.success')     
+	      redirect_to medicines_path   
+	    else   
+	      flash[:error] = t('medicine.delete.failure')     
+	      render :destroy
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb]: What will this render?

File: app/controllers/purchase_order_controller.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  def index
+    @purchase_orders = PurchaseOrder.all    
+  end
+
+  def new
+    @purchase_order = PurchaseOrder.new
+  end
+
+  def addmed
+    @purchase_order = PurchaseOrder.find(params[:id])
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb]: You'll need to use `sequence_num` in purchase orders as well.

File: app/models/appointment.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 class Appointment < ApplicationRecord
+	sequenceid :hospital , :appointments
```
Comment (hamzashariq7) [inline app/models/appointment.rb]: Fix indentation for this in all models.

File: app/models/medicine.rb [hunk 7]
```diff
@@ -1,7 +1,19 @@
...
 	belongs_to :hospital
+
+	def self.search(pattern)
+      if pattern.blank?  # blank? covers both nil and empty string
+        all
+      else
+        where('name LIKE ?', "%#{pattern}%")
+      end
+    end
   default_scope { where(hospital_id: Hospital.current_id) }
```
Comment (hamzashariq7) [inline app/models/medicine.rb:10]: move this above the function, right after the `belongs_to` statement.

File: app/models/purchase_order.rb [hunk 8]
```diff
@@ -1,7 +1,34 @@
...
   belongs_to :admin
   has_many :purchase_details, dependent: :destroy
   has_many :medicines, through: :purchase_details
+  def add_medicine(medicine)
+  	if medicine.quantity>0
+      if medicine.update(quantity: medicine.quantity-1)
+        self.update(price: self.price+=medicine.price)   
+  	    purchase_details.create(quantity: 1, medicine: medicine, hospital: medicine.hospital)        
+  	  end
+  	else
```
Comment (hamzashariq7) [inline app/models/purchase_order.rb]: indentation.

File: app/controllers/purchase_order_controller.rb (hunk 5 above)
Comment (tariq-9484) [reply app/controllers/purchase_order_controller.rb]: I'm expilicitly passing the medicine_id in search params to search it this way. This won't cause an issue

Comment (tariq-9484) [reply app/controllers/purchase_order_controller.rb]: been using sequence_num for everything else

File: Gemfile [hunk 9]
```diff
@@ -16,7 +16,13 @@ gem 'webpacker', '= 4.3.0'
 # Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
 gem 'turbolinks', '= 5.2.1'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
+gem 'jquery-rails'
+gem 'jquery-ui-rails'
```
Comment (hamzashariq7) [inline Gemfile]: This is not required.

File: app/assets/javascripts/application.js [hunk 10]
```diff
@@ -0,0 +1,5 @@
+//= require jquery
```
Comment (hamzashariq7) [inline app/assets/javascripts/application.js]: Is this file required? I don't think it is being used anymore.

File: app/controllers/medicines_controller.rb [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  end
+
+  def new
+  end
+
+  def show
+  end
+
+  def search_pred
+    @medicines = Medicine.search(params[:q])
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:23]: what is `:q`? Use meaningful names.

File: app/controllers/medicines_controller.rb [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    else
+      flash[:notice] = t('medicine.search.success')   
+      respond_to do |format|
+        format.js{ render 'purchase_order/search' }
+      end
+    end
+  end
+
+  def create
+    @medicine.hospital=current_hospital
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb]: space between `=`

File: app/controllers/purchase_order_controller.rb [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+      flash[:notice] = t('purchase_order.addmed.success')  
+      respond_to do |format|
+        format.js{ render 'purchase_order/update_price' }
+      end            
+    else   
+      flash[:error] = t('purchase_order.addmed.failure')     
+    end 
+  end
+
+  def show
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:29]: Add `respond_to` html blocks in all empty actions.

File: app/controllers/medicines_controller.rb (hunk 11 above)
Comment (tariq-9484) [reply app/controllers/medicines_controller.rb:23]: :q is used by tokenInput to send the search field value. its supposed to happen using this key. i havent used this ley

File: app/javascript/packs/application.js [hunk 14]
```diff
@@ -27,7 +26,9 @@ const imagePath = (name) => images(name, true)
 //
 // const images = require.context('../images', true)
 // const imagePath = (name) => images(name, true)
-
+//= require jquery
+//= require jquery_ujs
```
Comment (hamzashariq7) [inline app/javascript/packs/application.js]: Is this required?

File: app/models/appointment.rb (hunk 6 above)
Comment (hamzashariq7) [inline app/models/appointment.rb]: Fix indentation for this in all models.

File: app/javascript/packs/application.js (hunk 14 above)
Comment (tariq-9484) [reply app/javascript/packs/application.js]: removed

File: app/models/appointment.rb (hunk 6 above)
Comment (tariq-9484) [reply app/models/appointment.rb]: this is not even my model:D

Comment (tariq-9484) [reply app/models/appointment.rb]: done

File: app/models/purchase_order.rb [hunk 15]
```diff
@@ -1,7 +1,34 @@
...
+  sequenceid :hospital , :purchase_orders
+  validates :vendorname, presence: { message: "must be given" }
+  validates :price, presence: true, numericality: {:greater_than => -1}
+  validates :state, presence: true 
   belongs_to :hospital
   belongs_to :admin
   has_many :purchase_details, dependent: :destroy
   has_many :medicines, through: :purchase_details
   default_scope { where(hospital_id: Hospital.current_id) }
+  def add_medicine(medicine, quantity1)
```
Comment (hamzashariq7) [inline app/models/purchase_order.rb]: `quantity1 -> quantity_added`

File: app/views/layouts/application.html.erb [hunk 16]
```diff
@@ -5,10 +5,11 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
     <%= stylesheet_pack_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
-
-  </head>
-
+    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %> 
+    <%#= stylesheet_link_tag 'https://cdn.datatables.net/s/dt/dt-1.10.10,r-2.0.0/datatables.min.css' %>
```
Comment (hamzashariq7) [inline app/views/layouts/application.html.erb]: Remove the commented out tags.

File: app/views/medicines/index.html.erb [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+    <% @medicine.each do |medicine| %>   
+    <tr>   
+      <td><%= link_to medicine.name, medicine_path(medicine) %></td>   
+      <td><%= medicine.price %></td>   
+      <td><%= medicine.quantity%></td> 
+      <div>   
+       <td>   
+        <%= link_to 'Edit', edit_medicine_path(medicine) %> 
+        <%= link_to 'Delete', medicine_path(medicine), method: :delete %>   
+      </div>
```
Comment (hamzashariq7) [inline app/views/medicines/index.html.erb]: why is the div closing before the td?

File: app/views/medicines/index.html.erb [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+       <td>   
+        <%= link_to 'Edit', edit_medicine_path(medicine) %> 
+        <%= link_to 'Delete', medicine_path(medicine), method: :delete %>   
+      </div>   
+    </td>        
+  </tr>   
+  <% end %>   
+</tbody>   
+</table>  
+</div>
```
Comment (hamzashariq7) [inline app/views/medicines/index.html.erb]: Fix indentation and tag order in this file.

File: config/routes.rb [hunk 19]
```diff
@@ -1,5 +1,20 @@
 Rails.application.routes.draw do
 
+  resources :doctors
+  resources :patients
```
Comment (hamzashariq7) [inline config/routes.rb:4]: Remove this, it is a duplicate.

File: db/migrate/20200909123349_add_devise_to_users.rb [hunk 20]
```diff
@@ -4,7 +4,7 @@ class AddDeviseToUsers < ActiveRecord::Migration[6.0]
   def self.up
     change_table :users do |t|
       ## Database authenticatable
-      t.string :email,              null: false, default: ""
+      #t.string :email,              null: false, default: ""
```
Comment (hamzashariq7) [inline db/migrate/20200909123349_add_devise_to_users.rb]: Remove this comment.

File: .gitignore [hunk 21]
```diff
@@ -34,5 +34,5 @@
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
-
-/db/schema.rb
+db/schema.rb $ git rm --cached db/schema.rb rm db/schema.rb $ git commit -m Start ignoring schema.rb
```
Comment (hamzashariq7) [inline .gitignore:37]: Verify this.

Comment (tariq-9484) [reply .gitignore:37]: both of them do the same thing
doesnt matter which one you remove, so i removed the first one
